app/src/main/java/com/example/keniphhomemaintenance/navigation/Navigation.kt

```diff
@@ -0,0 +1,34 @@
+package com.example.keniphhomemaintenance.navigation
+
+import androidx.compose.runtime.Composable
+import androidx.navigation.NavType
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import androidx.navigation.navArgument
+import com.example.keniphhomemaintenance.dwelling_screen.DwellingScreen
+import com.example.keniphhomemaintenance.maintenance_screen.MaintenanceScreen
+
+@Composable
+fun Navigation() {
+    val navController = rememberNavController()
+    NavHost(navController = navController, startDestination = Screen.DwellingCreationScreen.route) {
+        composable(route = Screen.DwellingCreationScreen.route) {
+            DwellingScreen(navController = navController)
+            
+        }
+        composable(
+            route = Screen.MaintenanceListScreen.route + "/{dwellingName}",
+            arguments = listOf(
+                navArgument("dwellingName") {
+                    type = NavType.StringType
+                    nullable = true
+                }
+            )
+        ) {
+            MaintenanceScreen()
```

Comment:
Don't you need to pass in the arguments here?

Reply:
I don't believe so. Pretty sure arguments are passed in via the navController. See DwellingView.kt line 39. @NickPerozzi should have handled passing in the arguments in his PR.

Reply:
This will be one of the places where we will have to jigsaw fit our two parts in